feature/issue 1582

tggoo opened the pull request:
Fixes #1582.

This feature implements support for including a filename when adding parts to a multipart body.

tggoo commented:
@microsoft-github-policy-service agree

baywet commented on packages/abstractions/src/multipartBody.ts:
```diff
@@ -26,9 +26,10 @@ export class MultipartBody implements Parsable {
 	 * @param serializationCallback the serialization callback to use when serializing the part.
 	 */
-	public addOrReplacePart<T>(partName: string, partContentType: string, content: T, serializationCallback?: ModelSerializerFunction<Parsable>): void {
+	public addOrReplacePart<T>(partName: string, partContentType: string, content: T, fileName?: string, serializationCallback?: ModelSerializerFunction<Parsable>): void {
```
```suggestion
	public addOrReplacePart<T>(partName: string, partContentType: string, content: T, serializationCallback?: ModelSerializerFunction<Parsable>, fileName?: string): void {
```

To align with the dotnet implementation https://github.com/microsoft/kiota-dotnet/blob/dbbd3d0feddf0aeeb291de22f64f11d7778cd1e6/src/abstractions/MultipartBody.cs#L36

baywet reviewed (changes requested):
Thanks for the contribution!

baywet commented:
@tggoo a gentle reminder on this one.
Let us know if you have any additional comments or questions.

tggoo commented:
@baywet 
@andrueastman 

Apologies for the delay... I was caught up with work and studies. Thank you for your patience!

I’ve now finished addressing your feedback and made the requested changes. Please take a look when you have time. Let me know if anything else needs to be improved!

Thanks again!

andrueastman reviewed (approved):
Thanks for the contribution here @tggoo

baywet reviewed (approved):
Thank you for making the changes!
